File: P12977/oognuyh.java
package P12977;

import java.util.Arrays;

public class oognuyh {
    
    public int solution(int[] nums) {
        boolean[] isPrime = new boolean[1000 + 999 + 998 + 1];
        int numOfPrimes = 0;

        Arrays.fill(isPrime, true);

        isPrime[0] = isPrime[1] = false;

        for (int i = 2; i * i < isPrime.length; i++) 
            if (isPrime[i]) 
                for (int j = 2; i * j < isPrime.length; j++) isPrime[i * j] = false;

        for (int i = 0; i < nums.length; i++) 
            for (int j = i + 1; j < nums.length; j++) 
                for (int k = j + 1; k < nums.length; k++) 
                    if (isPrime[nums[i] + nums[j] + nums[k]]) numOfPrimes++;

        return numOfPrimes;
    }
}

File: P12977/oognuyh.kt
package P12977;

import kotlin.math.sqrt;

class oognuyh {

    fun solution(nums: IntArray): Int {
        var numOfPrimes = 0;
        val isPrime = BooleanArray(nums.maxOrNull()!! * 3 + 1) { true };

        (2 .. sqrt(isPrime.lastIndex.toDouble()).toInt())
            .filter { number -> isPrime[number] }
            .forEach { number -> (2 * number .. isPrime.lastIndex step number)
                .forEach { multiple -> isPrime[multiple] = false } };

        (0 .. nums.lastIndex)
            .forEach { i -> (i + 1 .. nums.lastIndex)
                .forEach { j -> (j + 1 .. nums.lastIndex)
                    .forEach { k -> if (isPrime[nums[i] + nums[j] + nums[k]])  numOfPrimes++ } } };
        
        return numOfPrimes;
    }
}